### PR TITLE
Checking elements in filter regardless of class order

### LINKED ISSUE #727: Styles combo disabled when using style with multiple, not sorted classes
## Are you reporting a feature request or a bug?

Bug

## [Check if the issue is already reported](http://docs.ckeditor.com/#!/guide/dev_issues_readme-section-avoid-duplicates)

https://dev.ckeditor.com/ticket/13206

## Provide detailed reproduction steps (if any)

1. Open: https://jsfiddle.net/z90dfv8n/
2. Click on the styles combo.

### Alternative steps

1. Create a minimal CKEditor instance that includes the `stylescombo` plugin and the `styleSet` configuration option set to:

```
[ { name: 'Marker', element: 'span', attributes: { 'class': 'b-test a-test' } } ]
```
2. Click on the styles combo.

### Expected result

Styles combo shows one style that's named "Marker".

### Actual result

Styles combo is empty and disables itself upon clicking on it.

## Other details

* With the class order switched (sorted lexically) the bug does not occur.
* The (bug-inducing) style is recognized but in this form the ACF filters it out from the styles list.
* Pasting this snippet: 

```
<p><span class="b-test a-test">Test</span></p>
```

into the source and switching modes back and forth produces a sorted output like this: 

```
<p><span class="a-test b-test">Test</span></p>
```

* Browser: All
* OS: All
* CKEditor version: 4.3
* Installed CKEditor plugins: `stylescombo`

### PR DESCRIPTION
**Pull request by msamsel**
## What is the purpose of this pull request?

Bug fix

## Does your PR contain necessary tests?

All patches which change the editor code must include tests. You can always read more
on [PR testing](https://docs.ckeditor.com/ckeditor4/docs/#!/guide/dev_contributing_code-section-tests),
[how to set the testing environment](https://docs.ckeditor.com/ckeditor4/docs/#!/guide/dev_tests) and
[how to create tests](https://docs.ckeditor.com/ckeditor4/docs/#!/guide/dev_tests-section-creating-your-own-test)
in the official CKEditor documentation.

### This PR contains

- [x] Unit tests
- [x] Manual tests

## What changes did you make?

- ~pretty much I've made change as it was in #728 and add manual tests~
- ~I couldn't re-use #728 :( there is unknown source repository, so I wasn't able to start from that PR.~
- I propose another solution which sorts class during check and do not change users markup

Close #727

### CODE REVIEW

File: tests/core/filter/manual/classcheckinlexicalorder.html [hunk 1]
```diff
@@ -0,0 +1,19 @@
+<textarea id="classic"><p>Hello world</p></textarea>
+<div id="inline" contenteditable="true"><p>Hello world</p></div>
+
+<script>
+	CKEDITOR.replace( 'classic', {
+		stylesSet: [
```
Comment (engineering-this) [inline tests/core/filter/manual/classcheckinlexicalorder.html]: When each pair of key and value is in new line it is much more easier to read. Also I'd consider extracting styleSet to variable, as it is used more than once.

File: tests/core/filter/manual/classcheckinlexicalorder.md [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+3. Check what is displayed in styles combo
+
+**Expected:** Styles combo should display 3 different styles
+
+----
+
+1. Select entire paragraph
+2. Apply style `Marker 2` to the paragraph
+3. Do not change selection or reselect the same text
+3. Re-open stylescombo
```
Comment (engineering-this) [inline tests/core/filter/manual/classcheckinlexicalorder.md]: Typo: step '3.' is twice. You could just use '1.' in every step and let markdown parser to create correct lists.

File: tests/core/filter/manual/classcheckinlexicalorder.md [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+1. Select paragraph
+2. Open stylecombo
+3. Check what is displayed in styles combo
+
+**Expected:** Styles combo should display 3 different styles
+
+----
+
+1. Select entire paragraph
+2. Apply style `Marker 2` to the paragraph
```
Comment (engineering-this) [inline tests/core/filter/manual/classcheckinlexicalorder.md]: I'd add explicit information to use the styles combo to apply certain style.
Eg. 'Apply style `Marker 2` from styles combo'

**Comment by msamsel** [comment]
> I think the problem is more general, than just disabled styles combo. This is just case when the issue is visible, but our approach should be wider, to cover all cases with multiple classes.

In my opinion is not the scope of this issue. It's more likely feature request where we start support classes without lexical order in entire editor. Currently there are some situation when editor's content is destroyed and recreated from the scratch. During this process class are re-created alphabetically. That's why to support described cases we should store somehow order of the classes and properly restore it. Comparing this to issue that style cannot be obtain from styles combo is much more wider task.
This conclusions are also somehow related to issues: #1671, #989.

@Comandeer WDYT about going into general and wider approach with this issue?

**Comment by msamsel** [comment]
I've rebased branch and change target to master.

**Comment by Comandeer** [comment]
> WDYT about going into general and wider approach with this issue?

It'd be nice, but seems like to big change without really visible advantages. IMO this PR should fix only the rich combo issue.

**Pull request merged**